openpype/plugins/publish/integrate.py

```diff
@@ -0,0 +1,835 @@
+
+
+class IntegrateAsset(pyblish.api.InstancePlugin):
+    """Register publish in the database and transfer files to destinations.
+
+    Steps:
+        1) Register the subset and version
+        2) Transfer the representation files to the destination
+        3) Register the representation
+
+    Requires:
+        instance.data['representations'] - must be a list and each member
+        must be a dictionary with following data:
+            'files': list of filenames for sequence, string for single file.
+                     Only the filename is allowed, without the folder path.
+            'stagingDir': "path/to/folder/with/files"
+            'name': representation name (usually the same as extension)
+            'ext': file extension
+        optional data
+            "frameStart"
+            "frameEnd"
+            'fps'
+            "data": additional metadata for each representation.
+    """
+
+    label = "Integrate Asset"
+    order = pyblish.api.IntegratorOrder
+    hosts = ["maya"]
+    families = ["workfile",
+                "pointcache",
+                "camera",
+                "animation",
+                "model",
+                "mayaAscii",
+                "mayaScene",
+                "setdress",
+                "layout",
+                "ass",
+                "vdbcache",
+                "scene",
+                "vrayproxy",
+                "vrayscene_layer",
+                "render",
+                "prerender",
+                "imagesequence",
+                "review",
+                "rendersetup",
+                "rig",
+                "plate",
+                "look",
+                "audio",
+                "yetiRig",
+                "yeticache",
+                "nukenodes",
+                "gizmo",
+                "source",
+                "matchmove",
+                "image",
+                "assembly",
+                "fbx",
+                "textures",
+                "action",
+                "harmony.template",
+                "harmony.palette",
+                "editorial",
+                "background",
+                "camerarig",
+                "redshiftproxy",
+                "effect",
+                "xgen",
+                "hda",
+                "usd",
+                "staticMesh",
+                "skeletalMesh",
+                "mvLook",
+                "mvUsd",
+                "mvUsdComposition",
+                "mvUsdOverride",
+                "simpleUnrealTexture"
+                ]
+    exclude_families = ["clip", "render.farm"]
+    default_template_name = "publish"
+
+    # Representation context keys that should always be written to
+    # the database even if not used by the destination template
+    db_representation_context_keys = [
+        "project", "asset", "task", "subset", "version", "representation",
+        "family", "hierarchy", "username"
+    ]
+
+    # Attributes set by settings
+    template_name_profiles = None
```

Comment:
If we'll change place where publish templates are defined the we should move it from publish plugins to separated part so it's not needed to know plugin name to receive it when it's needed elsewhere.